coalib/settings/ConfigurationGathering.py

```diff
@@ -132,6 +133,20 @@ def warn_config_absent(sections, argument, log_printer):
     return False
 
 
+def get_default_section(section, sections):
+    default_section = ('.'.join(section.split('.')[:-1])
+                       if unescaped_search_for('.', section) else None)
```

Comment:
I think it's better if we just disallow `\.`s in section names. Certainly improves readability and I can't think of a use case where a section name would need a `.`